deepchecks/checks/performance/class_performance_imbalance.py

```diff
@@ -0,0 +1,236 @@
+
+        Args:
+            metrics: alternative metrics to execute
+
+        Raises:
+            ValueError:
+                if provided dict of metrics is emtpy;
+                if one of the entries of hte provided metrics dict contains not callable value;
+        """
+        super().__init__()
+        self.alternative_metrics = metrics
+
+        if self.alternative_metrics is not None:
+
+            if len(self.alternative_metrics) == 0:
+                raise ValueError('Expected to receive not empty dict of callables!')
+
+            incorrect_argument = next(
+                (it for it in self.alternative_metrics.values() if not callable(it)),
+                None
+            )
+
+            if incorrect_argument is not None:
+                raise ValueError(
+                    "metrics - expected to receive 'Dict[str, MetricFunc]' but got " #pylint: disable=inconsistent-quotes
+                    f"'Dict[str, {type(incorrect_argument).__name__}]'!" #pylint: disable=inconsistent-quotes
+                )
+
+    def run(
+        self,
+        dataset: Dataset,
+        model: t.Any # TODO: find more precise type for model
+    ) -> CheckResult:
+        """Run Check.
+
+        Args:
+            dataset (Dataset): a dataset object
+            model (BaseEstimator): A scikit-learn-compatible fitted estimator instance
+
+        Returns:
+            CheckResult: value is a dictionary in format {'label': {'metric name': <value>, ...}, ...}
+
+        Raises:
+            DeepchecksValueError:
+                if task type is not binary or multi-class;
+                if dataset does not have a label column or the label dtype is unsupported;
+                if provided dataset is empty;
+        """
+        return self._class_performance_imbalance(dataset, model)
+
+    def _class_performance_imbalance(
+        self,
+        dataset: Dataset,
+        model: t.Any # TODO: find more precise type for model
+    ) -> CheckResult:
+        check_name = type(self).__name__
+        expected_model_types = [ModelType.BINARY, ModelType.MULTICLASS]
+
+        dataset.validate_label(check_name)
+        Dataset.validate_dataset(dataset, check_name)
+        task_type_validation(model, dataset, expected_model_types, check_name)
+
+        features = dataset.features_columns()
+
+        if features is None:
+            raise DeepchecksValueError(f'Check {check_name} requires dataset with features!')
+
+        y_true = t.cast(pd.Series, dataset.label_col())
+        y_pred = model.predict(features)
+
+        if self.alternative_metrics is not None:
+            df = pd.DataFrame.from_dict(self._execute_alternative_metrics(y_true, y_pred))
+
+        else:
+            labels = unique_labels(y_true, y_pred)
+            df = pd.DataFrame.from_dict({
+                name: dict(zip(
+                    labels,
+                    t.cast(t.Iterable, metric_func(y_true, y_pred, labels=labels))
```

Comment:
In order to function similarly to TrainTestDifferenceOverfit, better to have the user supplied metric be either scorer strings or sklearn scorers. Look how it's done there (and feel free to offer a different opinion)

Reply:
updated